### DISCUSSION TITLE
read of csv

### DISCUSSION THREAD

**Discussion by MarinCronograph** [gempy-project/gempy#936] [General]
hi,

I would luke to know how do i read of .csv given on  lines . ?

path_to_orientations=data_path + "/data/input_data/getting_started/simple_fault_model_orientations.csv",
        path_to_surface_points=data_path + "/data/input_data/getting_started/simple_fault_model_points.csv",

on tutorial web page 
https://docs.gempy.org/tutorials/ch1_fundamentals/ch1_1_basics.html#sphx-glr-tutorials-ch1-fundamentals-ch1-1-basics-py

I tried printing, in python,  but it doesnt work

regards

**Comment by javoha** [comment]
Hi,
what exactly did not work? The data for the tutorial models is stored in another github repository. The defined path consists of two parts: The first is the repo `(data_path = 'https://raw.githubusercontent.com/cgre-aachen/gempy_data/master/'`, the second is the location in the repo (`/data/input_data/getting_started/simple_fault_model_orientations.csv`). 
Just running the following code should give you the complete path:

```
data_path = 'https://raw.githubusercontent.com/cgre-aachen/gempy_data/master/'
print(data_path+data_path + "/data/input_data/getting_started/simple_fault_model_orientations.csv")
```
You could also download the files and reference them in your local directory.

Does this answer your question?

Cheers
Jan

**Reply by clearlove75** [reply]
Hi Jan，
I wrote an e-mail and I fellow you on github. You're a friendly guy who likes to help members. Recently I was having problems trying to create a geological model with gempy, but it kept coming up with all kinds of errors, and I thought I might need a sample code for reference. Could you simply write a piece of code to generate a geological interface that contains random geological events.
My version of gempy is 2024.2.0.2 and python is 3.12.4.
By the way, here is my code, which still reports errors despite all my modifications.I really hope you can reply to me.

**Reply by MarinCronograph** [reply]
hi @javoha ,

I meant  like printing the content  on 
https://raw.githubusercontent.com/cgre-aachen/gempy_data/master/data/input_data/getting_started/simple_fault_model_orientations.csv and that would be

X,Y,Z,azimuth,dip,polarity,formation
1000,1000,300,90,18.435,1,Shale
400,1000,420,90,18.435,1,Sandstone_2
500,1000,300,270,71.565,1,Main_Fault

It would be, i guess , scraping the data the printing . Just wanted to do this  mora automatic, in python and not manually combining like I actually did for  (URL)  https://raw.githubusercontent.com/cgre-aachen/gempy_data/master/data/input_data/getting_started/simple_fault_model_orientations.csv 

rgds

**Reply by javoha** [reply]
Hi,
if I understand you correctly you want to print the dataframe directly in your python environment? You are correct that gempy does not have a pretty print function for these tables at the moment (which we should fix). You access the orientation file using `geo_model.orientations_copy` but it contains the orientations converted to pole vectors and lacks the formation names.
You could also just read the tables using pandas (`pd.read_csv()`) and print the resulting dataframe. 
What exactly do you want to do with it. Maybe I can give better hints then.
Cheers
Jan

**Reply by MarinCronograph** [reply]
hi @javoha 

I just wanted to have control over the data nad see it/print from pytohn, that is all.

I managed to see the data with  print(geo_model.surface_points_copy), under

[( 700., 1000., 300.  ,   3719548, 0.),
 ( 600., 1000., 200.  ,   3719548, 0.),
 ( 500., 1000., 100.  ,   3719548, 0.),
 (1000., 1000., 600.  ,   3719548, 0.),
 (1100., 1000., 700.  ,   3719548, 0.),
 (1000.,   50., 350.  , 170628891, 0.),

However tried with pd.read_csv(_**inputted the path here but it didnt work**_).  Can you help with this?

rgds

Marin

**Comment by javoha** [comment]
Hi,
have you tried this:

```
import pandas as pd

df = pd.read_csv(
    'https://raw.githubusercontent.com/cgreaachen/gempy_data/master//data/input_data/getting_started/simple_fault_model_orientations.csv')

print(df)
```
This saves the table as a pandas dataframe (called _df_ in my example) and you can do whatever you want with it.

Cheers,
Jan

**Reply by MarinCronograph** [reply]
hi @javoha 

i managed to print points (df) , i guess these are base  points of a layer and faults(simple_fault_model_points.csv)

   Unnamed: 0     X     Y           Z    formation
0            0  1000    50  450.000000        Shale
1            1  1000   150  433.333333        Shale
...... and so on.

many thanks

Marin Mirosevic
